apisix/plugins/lago.lua

```diff
@@ -0,0 +1,224 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local type         = type
+local pairs        = pairs
+local math_random  = math.random
+local ngx          = ngx
+
+local http            = require("resty.http")
+local bp_manager_mod  = require("apisix.utils.batch-processor-manager")
+local core            = require("apisix.core")
+local str_format      = core.string.format
+
+local plugin_name = "lago"
+local batch_processor_manager = bp_manager_mod.new("lago logger")
+
+local schema = {
+    type = "object",
+    properties = {
+        -- core configurations
+        endpoint_addrs = {
+            type = "array",
+            minItems = 1,
+            items = core.schema.uri_def,
+            description = "Lago API address, like http://127.0.0.1:3000, "
+                        .. "it supports both self-hosted and cloud. If multiple endpoints are"
+                        .. " configured, the log will be pushed to a randomly determined"
+                        .. " endpoint from the list.",
+        },
+        endpoint_uri = {
+            type = "string",
+            minLength = 1,
+            default = "/api/v1/events/batch",
+            description = "Lago API endpoint, it needs to be set to the batch send endpoint.",
+        },
+        token = {
```

Comment:
I think it can be set as an API `encrypt_fields`.

Reply:
fixed